cloud/amazon/ec2_vpc_vgw.py

```diff
@@ -0,0 +1,561 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+module: ec2_vpc_vgw
+short_description: Create and delete AWS VPN Virtual Gateways.
+description:
+  - Creates AWS VPN Virtual Gateways
+  - Deletes AWS VPN Virtual Gateways
+  - Attaches Virtual Gateways to VPCs
+  - Detaches Virtual Gateways from VPCs
+version_added: "2.1"
+requirements: [ boto3 ]
+options:
+  state:
+    description:
+        - present to ensure resource is created.
+        - absent to remove resource
+    required: false
+    default: present
+    choices: [ "present", "absent"]
+  name:
+    description:
+        - name of the vgw to be created or deleted
+    required:
+        - true when combined with a state of 'present'
+        - false when combined with a state of 'absent'
+  type:
+    description:
+        - type of the virtual gateway to be created
+    required:
+        - true when combined with a state of 'present'
+        - false when combined with a state of 'absent'
+  vpn_gateway_id:
+    description:
+        - vpn gateway id of an existing virtual gateway
+    required: false
+  vpc_id:
+    description:
+        - the vpc-id of a vpc to attach or detach
+    required: false
+  wait_timeout:
+    description:
+        - number of seconds to wait for status during vpc attach and detach
+    required: false
+    default: 320
+  tags:
+    description:
+        - dictionary of resource tags of the form: { tag1: value1, tag2: value2 }
+    required: false
+author: Nick Aslanidis (@naslanidis)
+extends_documentation_fragment: aws
```

Comment:
Missing ec2 fragment for region - I need to fix my own modules for this as well.

I believe it should read as: 

```
extends_documentation_fragment:
    - aws
    - ec2
```